core/src/smartcontracts/mod.rs

```diff
@@ -38,7 +38,7 @@ pub trait Evaluate<W: WorldTrait> {
 
 /// This trait should be implemented for all Iroha Queries.
 #[allow(clippy::missing_errors_doc)]
-pub trait Query<W: WorldTrait>: QueryOutput {
+pub trait ExecutableQuery<W: WorldTrait>: Query {
```

Comment:
I am open to more appropriate naming of `ExecutableQuery`, though I would stick to `Query` vs `QueryOutput` because that's exposed in the client

Reply:
`ValidQuery`? 
`Inquiry`? 
`Qry`